deps/dicer/lib/HeaderParser.js

```diff
@@ -72,39 +72,37 @@ HeaderParser.prototype._parseHeader = function() {
   if (this.npairs === this.maxHeaderPairs)
     return;
 
-  var lines = this.buffer.split(RE_CRLF), len = lines.length, m, h,
-      modded = false;
+  const lines = this.buffer.split(RE_CRLF), 
+    len = lines.length;
+  let m, h;
 
-  for (var i = 0; i < len; ++i) {
+  for (let i = 0; i < len; ++i) {
     if (lines[i].length === 0)
       continue;
     if (lines[i][0] === '\t' || lines[i][0] === ' ') {
       // folded header content
       // RFC2822 says to just remove the CRLF and not the whitespace following
       // it, so we follow the RFC and include the leading whitespace ...
-      this.header[h][this.header[h].length - 1] += lines[i];
-    } else {
-      m = RE_HDR.exec(lines[i]);
-      if (m) {
-        h = m[1].toLowerCase();
-        if (m[2]) {
-          if (this.header[h] === undefined)
-            this.header[h] = [m[2]];
-          else
-            this.header[h].push(m[2]);
-        } else
-          this.header[h] = [''];
-        if (++this.npairs === this.maxHeaderPairs)
-          break;
-      } else {
-        this.buffer = lines[i];
-        modded = true;
-        break;
+      if (h) {
```

Comment:
What was the intention of this rewrite?

Reply:
I think this is the patch?

Reply:
Do you see a potential regression?

Reply:
I don't understand neither the original code, nor the fix, which is why the change makes me pause. I guess I would need to reread carefully both the original and the change, to fully understand what is happening in both scenarios.

Reply:
https://www.rfc-editor.org/rfc/rfc2616#section-2.2
```
   HTTP/1.1 header field values can be folded onto multiple lines if the
   continuation line begins with a space or horizontal tab. All linear
   white space, including folding, has the same semantics as SP. A
   recipient MAY replace any linear white space with a single SP before
   interpreting the field value or forwarding the message downstream.
```

And for this part there is the test "Folded values" in "dicer-headerparser". If it was wrong the test there would fail.